### PR TITLE
cpuid: replace unmaintained intel-go/cpuid with fork

### PR DESCRIPTION
**Pull request by leongross**
Replace intel-go/cpuid with u-root fork [u-root/cpuid](github.com/u-root/cpuid). The fork is a copy of u-roots in-tree vendored modules with additional tinygo build support by utilizing CGO.

In prior tinygo adaptions for u-root, the package `pkg/cpuid/` (#3077) has been used. But this package only provides functionality for `cpuid_low`, not the other features.

For completeness sake, it is important to mention, that this package only solves the build issues for a certain set of problems:

* when intel-go-cpuid was used
* on x86_64 platforms

Cases left out:

* usage of vendor/golang.org/x/sys/cpu.cpuid
* usage of klauspost/cpuid

### CODE REVIEW

File: cmds/core/sluinit/uinit_linux.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// Copyright 2019 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline cmds/core/sluinit/uinit_linux.go:1]: this seems unrelated to cpuid?

**Review by rminnich — DISMISSED** [review]

File: pkg/msr/msr_linux_amd64.go [hunk 2]
```diff
@@ -274,13 +274,9 @@ func (m MSR) TestAndSet(c CPUs, clearMask uint64, setMask uint64) []error {
...
 func Locked() error {
-	vendor, err := cpuid.CPUManufacturerID()
-	if err != nil {
-		return err
-	}
-
+	vendor := cpuid.VendorIdentificatorString
 	// TODO: support more than Intel. Use the vendor id to look up msrs.
-	if vendor != cpuid.ManufacturerIDIntel {
+	if vendor != "GenuineIntel" {
```
Comment (binjip978) [inline pkg/msr/msr_linux_amd64.go:279]: maybe it's better to have a constant for "GenuineIntel"?

**Review by binjip978 — DISMISSED** [review]

File: cmds/core/sluinit/uinit_linux.go (hunk 1 above)
Comment (leongross) [reply cmds/core/sluinit/uinit_linux.go:1]: Yes, it is. Since it is not really important I will remove this from the PR.

File: pkg/msr/msr_linux_amd64.go (hunk 2 above)
Comment (leongross) [reply pkg/msr/msr_linux_amd64.go:279]: This was a constant that I newly introduced in the in-tree cpuid package. Since it is only really used in the msr command and there just string literal `"GenuineIntel"` is checked I think it is not necessary to abstract this.

**Review by binjip978 — APPROVED** [review]
